tests/ssg_test_suite/virt.py

```diff
@@ -158,3 +159,36 @@ def start_domain(domain):
         domain.create()
         logging.debug('Waiting 30s for domain to start')
         time.sleep(30)
+
+
+def reboot_domain(domain, domain_ip, ssh_port):
+    timeout = 120           # Timeout for domain shutdown and boot.
+    connection_timeout = 5  # Timeout on the socket before attempting to connect.
+
+    logging.debug("Shutting down domain '{0}'".format(domain.name()))
+    domain.shutdown()
+
+    # Wait until domain shuts down.
+    logging.debug("Waiting for domain to shutdown")
+    start_time = time.perf_counter()
+    while domain.isActive():
+        time.sleep(1)
+        if time.perf_counter() - start_time >= timeout:
+            raise TimeoutError("Timeout reached: '{0}' domain failed to shutdown."
+                               .format(domain.name()))
```

Comment:
I'm not a fan of the traceback it prints:
```
INFO - Rebooting domain 'rhel8.2-night' before final scan.
Traceback (most recent call last):
  File "test_suite.py", line 323, in <module>
    main()
  File "test_suite.py", line 319, in main
    options.func(options)
  File "/home/wsato/git/content/tests/ssg_test_suite/profile.py", line 43, in perform_profile_check
    checker.test_target(options.target)
  File "/home/wsato/git/content/tests/ssg_test_suite/oscap.py", line 621, in test_target
    self._test_target(target)
  File "/home/wsato/git/content/tests/ssg_test_suite/profile.py", line 22, in _test_target
    self.run_test_for_all_profiles(profiles)
  File "/home/wsato/git/content/tests/ssg_test_suite/oscap.py", line 635, in run_test_for_all_profiles
    self._run_test(profiles[0], test_data)
  File "/home/wsato/git/content/tests/ssg_test_suite/profile.py", line 33, in _run_test
    result = runner.run_stage(stage)
  File "/home/wsato/git/content/tests/ssg_test_suite/oscap.py", line 341, in run_stage
    result = self.final()
  File "/home/wsato/git/content/tests/ssg_test_suite/oscap.py", line 419, in final
    self.environment.reboot()
  File "/home/wsato/git/content/tests/ssg_test_suite/test_env.py", line 156, in reboot
    virt.reboot_domain(self.domain, self.domain_ip, self.ssh_port)
  File "/home/wsato/git/content/tests/ssg_test_suite/virt.py", line 178, in reboot_domain
    .format(domain.name()))
TimeoutError: Timeout reached: 'rhel8.2-night' domain failed to shutdown.
```

Someone should probably catch this exception.

Reply:
Can you elaborate more on this? Do you want the traceback to be hidden and to have just the error message printed? I think we should terminate testing in this case anyway.

Reply:
> Do you want the traceback to be hidden and to have just the error message printed?

Yes, UX of traceback is not nice. 

> I think we should terminate testing in this case anyway.

Agreed.